_posts/2021-06-29-high-level-graphs.md

```diff
@@ -0,0 +1,160 @@
+2. Implement cull method for ArrayOverlapLayer #7789 https://github.com/dask/dask/issues/7789
+
+## Slicing and high level graphs
+### Summary
+
+Profiling `map_overlap`, we saw that a lot of time is being spent in slicing operations. So, 
+
+Meanwhile, Rick Zamora has been working on the dataframe side of Dask, using high level graphs to improve dataframe slicing/selections.
+
+### Work to date
+
+A couple of minor bugfixes/improvements:
+1. Merged PR by Genevieve Buckley: [SimpleShuffleLayer should compare parts_out with set(self.parts_out) #7787](https://github.com/dask/dask/pull/7787)
+
+2. Merged PR by Genevieve Buckley: [Make Layer get_output_keys officially an abstract method #7775](https://github.com/dask/dask/pull/7775)
+
+### Ongoing work
+
+1. Rick Zamora: [[WIP] Add DataFrameGetitemLayer to simplify HLG Optimizations #7663](https://github.com/dask/dask/pull/7663)
+
+2. Genevieve Buckley: [Array slicing HighLevelGraph layer #7655](https://github.com/dask/dask/pull/7655) (Note: benchmarking indicates that this approach is not substantially faster than the current )
+
+
+## Visualization
+
+### Summary
+
+We've also put some work into making better visualizations for Dask objects (including high level graphs).
+
+Defining a `_repr_html_` method for your classes is a great way to get nice HTML output when you're working with jupyter notebooks. You can read [this post](http://matthewrocklin.com/blog/2019/07/04/html-repr) to see more neat HTML representations in other scientific python libraries.
+
+Dask already uses HTML representations in lots of places (like the `Array` and `Dataframe` classes). We now have new HTML representations for `HighLevelGraph` and `Layer` objects, as well as `Scheduler` and `Client` objects in Dask distributed.
+
+
+### Work to date
+
+1. Merged PR by Jacob Tomlinson: [Add HTML repr to scheduler_info and incorporate into client and cluster reprs #4857](https://github.com/dask/distributed/pull/4857)
+2. Merged PR by Jacob Tomlinson: [HTML reprs CLient.who_has & Client.has_what](https://github.com/dask/distributed/pull/4853)
+3. Merged PR by Genevieve Buckley: Implementation of HTML repr for HighLevelGraph layers #7763 https://github.com/dask/dask/pull/7763
+4. Merged PR by Genevieve Buckley [Automatically show graph visualization in jupyter notebooks #771](https://github.com/dask/dask/pull/7716)
+5. Merged PR by Genevivee Buckley: [Adding chunks and type information to dask high level graphs #7309](https://github.com/dask/dask/pull/7309). This PR inserts extra information into the high level graph, so that we can create richer visualizations using this extra context later on.
+
+### Example
+#### Before:
+```
+<dask.highlevelgraph.HighLevelGraph at 0x7f9851b7e4f0>
+```
+
+#### After (HTML representation):
+
+![Skeleton network of blood vessels within a healthy lung](/images/2021-highlevelgraph-html-repr.png)
```

Comment:
Interesting alt text?

Reply:
Sorry - I copy-pasted the image formatting from a previous post because I know the formatting that displays the image for me locally in VSCode is not the same as what displays once it's up online.

Thanks for catching this, Gabe